middleware/kubernetes/autopath.go

```diff
@@ -55,7 +55,9 @@ func (apw *AutoPathWriter) overrideMsg(res *dns.Msg, force bool) error {
 		if apw.original.Name == a.Header().Name {
 			continue
 		}
-		res.Answer = append(res.Answer, newCNAME(apw.original.Name, dns.Fqdn(a.Header().Name), a.Header().Ttl))
+		res.Answer = append(res.Answer, nil)
```

Comment:
Maybe this instead: https://codingair.wordpress.com/2014/07/18/go-appendprepend-item-into-slice/

?